### ISSUE TITLE
Evaluate generating multiple component models

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#224]
One model for choosing the product, one model to choose the "conflated component", one model to choose the component.

This might help increasing the accuracy (e.g. it's easier to tell that a bug belongs to Core::DOM rather than Core::DOM: Quota Manager).

**Comment by yixinsun** [comment]
I am thinking of something like 'hierarchical classification'. Did a bit of research and there are 2 ideas:
1) stacking: in my understanding, this is to stack classifiers at different levels [paper]((https://arxiv.org/abs/1807.08825))
2) Hierarchical neural network: model learned from the maximum posterior [paper](https://arxiv.org/pdf/1705.11105.pdf)

Are you thinking about something closer to the stacking model?

**Comment by marco-c** [comment]
Yes.

We have:
Level 1: Product (e.g. Core, Firefox, Toolkit, etc.)
Level 2: Component Group (e.g. Core::DOM*, Core::JavaScript*, etc.)
Level 3: Component

**Comment by marco-c** [comment]
We could also try https://scikit-learn.org/stable/modules/multiclass.html#multioutput-classification or https://scikit-learn.org/stable/modules/multiclass.html#classifier-chain.
